wgpu/src/lib.rs

```diff
@@ -816,7 +816,7 @@ impl Drop for ShaderModule {
 /// Any necessary shader translation (e.g. from WGSL to SPIR-V or vice versa)
 /// will be done internally by wgpu.
 #[non_exhaustive]
-pub enum ShaderSource<'a> {
+pub enum ShaderSource<#[cfg(any(feature = "spirv", feature = "glsl", feature = "wgsl"))] 'a> {
```

Comment:
Instead of this - let's have a dummy variant that is only enabled when all features are off. Right now - this technically means that adding features can cause code to break, which differs from the advice that features should only be additive

Reply:
Good idea, I decided to just always have the `Dummy` variant present and just slap a `#[doc(hidden)]` on it. I hope that's okay!